src/shipit_code_coverage/shipit_code_coverage/github.py

```diff
@@ -43,30 +43,24 @@ def post_github_status(self, commit_sha):
     def get_commit(self, mercurial_commit):
         def get_commit():
             r = requests.get('https://api.pub.build.mozilla.org/mapper/gecko-dev/rev/hg/%s' % mercurial_commit)
+            r.raise_for_status()
+            return r.text.split(' ')[0]
 
-            if r.ok:
-                return r.text.split(' ')[0]
-
-            return None
-
-        ret = wait_until(get_commit)
-        if ret is None:
+        try:
+            return retry(get_commit)
```

Comment:
Note that also the waiting times have to change. There are 5 retries with 30 seconds between them by default with `retry`. With `wait_until` the interval was the same (30 seconds), but the overall timeout was 30 minutes.
In the cases in this file, either use the same as before or even better same timeout but longer intervals (e.g. 2 minutes).

Reply:
Done in 4a557b8